### PR TITLE
feat(ports): implement additional ports management

### PR DESCRIPTION
**Pull request by lost-end-found**
This pull request introduces a new feature to manage additional port mappings for Traefik in a Docker Swarm environment. The main changes include adding a new component for managing ports, updating the server API to handle port configurations, and adding corresponding translations.

### New Feature: Manage Additional Ports for Traefik

* **Component Addition:**
  - [`apps/dokploy/components/dashboard/settings/web-server/manage-traefik-ports.tsx`](diffhunk://#diff-91ea20ec069355061c00c11dcc997293cc2423ca0c1420d5ae0994a961ecfb0eR1-R215): Added a new `ManageTraefikPorts` component that provides a modal interface for managing additional port mappings for Traefik. This component includes features to add, remove, and edit port mappings, and to persist port configurations through API calls.

* **Integration in UI:**
  - [`apps/dokploy/components/dashboard/settings/servers/actions/show-traefik-actions.tsx`](diffhunk://#diff-c46b9a41e0d23dbbff99fe5827683f41f78f6e3a12ebf3ad3128a4097a822100R29): Imported and integrated the `ManageTraefikPorts` component into the existing UI, allowing users to manage ports from the server settings dashboard. [[1]](diffhunk://#diff-c46b9a41e0d23dbbff99fe5827683f41f78f6e3a12ebf3ad3128a4097a822100R29) [[2]](diffhunk://#diff-c46b9a41e0d23dbbff99fe5827683f41f78f6e3a12ebf3ad3128a4097a822100R132-R139)

* **API Enhancements:**
  - [`apps/dokploy/server/api/routers/settings.ts`](diffhunk://#diff-ae03c4d7c9548d76636c5e63c9950fb188cc072ff5c57d76872074570e4caa91R71-R81): Added new API endpoints `updateTraefikPorts` and `getTraefikPorts` to handle updating and retrieving Traefik port configurations. These endpoints integrate with the Traefik setup process. [[1]](diffhunk://#diff-ae03c4d7c9548d76636c5e63c9950fb188cc072ff5c57d76872074570e4caa91R71-R81) [[2]](diffhunk://#diff-ae03c4d7c9548d76636c5e63c9950fb188cc072ff5c57d76872074570e4caa91R720-R776)

* **Traefik Setup Update:**
  - [`packages/server/src/setup/traefik-setup.ts`](diffhunk://#diff-c7ba7f17cb123f305f679bcd93556195954a10cfa0953ea18780adc7e4967e39R19-R30): Updated the `initializeTraefik` function to include additional ports in the Traefik service configuration. [[1]](diffhunk://#diff-c7ba7f17cb123f305f679bcd93556195954a10cfa0953ea18780adc7e4967e39R19-R30) [[2]](diffhunk://#diff-c7ba7f17cb123f305f679bcd93556195954a10cfa0953ea18780adc7e4967e39R93-R97)

* **Translations:**
  - [`apps/dokploy/public/locales/en/settings.json`](diffhunk://#diff-b2d23062842c4b343c92b1ec1cb2483af3d67d7a8b08f6e854a3213decedd729R21-R28): Added new translation strings for the port management feature, including labels and messages for the UI.
  
  
Screenshot 2024-12-18 at 16 48 05
Screenshot 2024-12-18 at 16 48 11
Screenshot 2024-12-18 at 16 48 19

### CODE REVIEW

**Comment by 190km** [comment]
it looks really good, but with a demo or images it could be even better!

**Comment by lost-end-found** [comment]
Updated with screenshots

File: apps/dokploy/server/api/routers/settings.ts [hunk 1]
```diff
@@ -68,6 +68,17 @@ import {
...
+	serverId: z.string().optional(),
+	additionalPorts: z.array(
+		z.object({
+			targetPort: z.number(),
+			publishedPort: z.number(),
+			publishMode: z.enum(["ingress", "host"]).default("host"),
+		}),
+	),
+});
+
```
Comment (Siumauricio) [inline apps/dokploy/server/api/routers/settings.ts]: Can we use this schema inside the input object from the mutation?

**Review by Siumauricio — CHANGES_REQUESTED** [review]
Looks too good! I left a small comment thanks!

**Pull request merged**